### DISCUSSION TITLE
Swipe View improvements

### DISCUSSION THREAD

**Discussion by nadiration** [Docile-Alligator/Infinity-For-Reddit#938] [Ideas]
I can't thank you @Docile-Alligator enough for adding this feature at my request #269 . There are some minor improvements I would like to be implemented inorder to make it more easier/accessible:

1. The Feature is buried under Settings > Gestures & Buttons which isn't easy to find or discover. It would be easier to move it to the constant three dots menu at the top right specially in Change Post Layout menu. That way it's easier to switch the post view without going to settings and losing your scroll position also it's easier for new users to discover this useful feature.
2. When you switch to Swipe mode the post layout should follow displaying one post at a time.
3. Swiping needs to continue working when you click on the post [image] to fully view. Leaving you clean image/post or meme with no title or other distractions (this is specifically useful for r/memes). See linked screenrecord for reference.

**Sync's implementation of the swipe view:** [**Video**](https://youtu.be/sMI9FEg_Byw)

**Comment by Docile-Alligator** [comment]
Thank you for these suggestions!

**Comment by nadiration** [comment]
@Docile-Alligator any plans for implementing this?

**Comment by rsmyst** [comment]
+1 on this. I really like the swipe to navigate on just the images like what boost for reddit has. Would also be more useful if it was actually in the post layout section rather than deep in the settings

also imo, swiping between post content should also be possible. For example, when viewing an image in a post and you swipe , it should navigate you to the image/video/link of the next post or just bring up the post if it's a text post. Having the image in fullscreen is much more immersive than seeing upvote, comment, etc. type of buttons along with it.

**Comment by jakenvac** [comment]
sorry for commenting on an older issue. I came here to make basically this request though so I thought best to not create a dupe.

Specifically swiping between media when using gallery mode would be really good. I hate to say it but a sort of tiktok style navigation would be awesome.

**Comment by Wladefant** [comment]
I think the implemented swipe mode is better than the one on Sync, because once you turn it on in infinity you can use it when you want and also use other functions from infinity without turning it off, like you would have to do in sync

**Comment by Wladefant** [comment]
> also imo, swiping between post content should also be possible. For example, when viewing an image in a post and you swipe , it should navigate you to the image/video/link of the next post or just bring up the post if it's a text post. Having the image in fullscreen is much more immersive than seeing upvote, comment, etc. type of buttons along with it.

@rsmyst I do not support everything you said but what should really be implemented is, that if i have an image in fullscreen mode and it is the last image in the post something should happen (would be cool of you could customize it in a dropdown menu). For example next image of next post, comments...

The dropdown menu should be somewhere in the post section?

**Comment by rsmyst** [comment]
Actually, I was in fact suggesting a swiping-to-next-post action once you've reached the last image of a gallery post while in fullscreen. Rather than abruptly ending, it could fluidly lead into the next post without exiting fullscreen and swiping over.

**Comment by Wladefant** [comment]
> Actually, I was in fact suggesting a swiping-to-next-post action once you've reached the last image of a gallery post while in fullscreen. Rather than abruptly ending, it could fluidly lead into the next post without exiting fullscreen and swiping over.

Yes, here i agree completely

@rsmyst if you want you can create a feature request only for this feature, so users can upvote and prioritize it

**Comment by nikhilCad** [comment]
Joey for Reddit also has a good implementation of this(install app and click on tv icon in the top). Though they have stuffed the upvotes and comments in bottom, but I think a layout like Instagram Reels(upvotes/comments on side, post title/subreddit name at bottom) will look better.

The way Boost for Reddit has implemented it is pretty good.
